### DISCUSSION TITLE
NodeJS version of Altair?

### DISCUSSION THREAD

**Discussion by mhkeller** [vega/altair#2972] [Ideas]
Hello, I wanted to see what your thoughts were on making a port of Altair for NodeJS. I saw you have one already for R. I do a lot of data cleaning and analysis in Node and I use Vega for some of the output. (I made [this library](https://github.com/mhkeller/plot) to simplify the generation of Vega, Observablehq/plot and plotly charts from Node.) 

The Altair API is really nice and simplified, though and it would be a great utility for folks working in Node to use. It would also be nice to not have to switch languages to use Vega with a less verbose API, since Vega is already JavaScript. I'd be happy to help lend a hand in building it, too.

**Comment by jheer** [comment]
We have a Vega-Lite JS API here: https://github.com/vega/vega-lite-api

Online notebook examples here: https://observablehq.com/collection/@vega/vega-lite-api

Admittedly the documentation could be better -- we simply piggyback off the content contained in the Vega-Lite JSON schema, from which the API is generated.

**Reply by mhkeller** [reply]
Got it. Thanks for the reference and for the work in building it!

**Reply by mhkeller** [reply]
I added support for vega-lite-api in my plotting tool (https://github.com/mhkeller/plot) and it's been really nice to use so far. Thanks again for the tip.
